packages/server/postgres/migrations/1627394777775_addPollsTable.ts

```diff
@@ -0,0 +1,58 @@
+import {MigrationBuilder, ColumnDefinitions} from 'node-pg-migrate'
+import {Client} from 'pg'
+
+import getPgConfig from '../getPgConfig'
+
+export const shorthands: ColumnDefinitions | undefined = undefined
+
+export async function up(pgm: MigrationBuilder): Promise<void> {
+  const client = new Client(getPgConfig())
+  await client.connect()
+
+  await client.query(`
+    DO $$
+    BEGIN
+      IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'PollStatusEnum') THEN
+        CREATE TYPE "PollStatusEnum" AS ENUM (
+          'creating',
+          'active',
+          'editing',
+          'ended'
+        );
+      END IF;
+    END
+    $$;
+
+    CREATE TABLE IF NOT EXISTS "Poll" (
+      "id" VARCHAR(100) PRIMARY KEY NOT NULL,
```

Comment:
What if we use https://www.postgresql.org/docs/current/datatype-uuid.html here?

Reply:
today we just use a snowflake as it's shorter than a UUID.
UUIDs work, they're just a little longer, so it's not very pretty if the entity is in the URL (or if we have to debug something).

The norm is `"id" INT GENERATED BY DEFAULT AS IDENTITY` and then just prefix the GraphQL ID with the name of the table. sequences do give away a little bit of data about how many records we have though. For example, a competitor could create a new poll, look at the ID & figure out approximately how popular the feature is. we could avoid that if we encoded it with a secret, but then we're back to the long IDs again

Reply:
Makes sense. Thank you for all the details! I'll change it to `"id" INT GENERATED BY DEFAULT AS IDENTITY` 👍